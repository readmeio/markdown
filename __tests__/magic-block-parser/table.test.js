const unified = require('unified');
const remarkParse = require('remark-parse');
const rehypeSanitize = require('rehype-sanitize');

const parser = require('../../processor/parse/magic-block-parser');
const options = require('../../options.json').markdownOptions;

const sanitize = { attributes: [] };
const process = (text, opts = options) =>
  text &&
  unified()
    .use(remarkParse, opts)
    .data('settings', { position: false })
    .use(parser.sanitize(sanitize))
    .use(rehypeSanitize)
    .parse(text);

describe('Table Blocks', () => {
  it('does not display data outside the declared dimensions', () => {
    const text = `[block:parameters]
    {
      "data": {
        "h-0": "Left",
        "h-1": "Center",
        "h-2": "Right",
        "0-0": "Left 0",
        "0-1": "Center 0",
        "0-2": "Right 0",
        "1-0": "Left 1",
        "1-1": "Center 1",
        "1-2": "Right 1",
        "2-0": "Left 2",
        "2-1": "Center 2",
        "2-2": "Right 2"
      },
      "cols": 2,
      "rows": 2
    }
    [/block]`;

    expect(process(text)).toMatchSnapshot();
  });
});
