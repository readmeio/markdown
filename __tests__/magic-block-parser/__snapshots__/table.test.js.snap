// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Blocks does not display data outside the declared dimensions 1`] = `
Object {
  "children": Array [
    Object {
      "align": Array [
        "left",
        "left",
      ],
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Left",
                },
              ],
              "type": "tableHead",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Center",
                },
              ],
              "type": "tableHead",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Left 0",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Center 0",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Left 1",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Center 1",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "root",
}
`;
